File: internal/model/tracking.go
package model

import "time"

type DistShipment struct {
	ID                  uint64     `gorm:"primaryKey" json:"id"`
	TenantID            uint64     `gorm:"index;not null" json:"tenantId"`
	DistOrderID                uint64     `gorm:"index;not null" json:"distOrderId"`
	ShipmentNo          string     `gorm:"size:32;not null" json:"shipmentNo"`
	Status              string     `gorm:"size:32;not null;default:pending" json:"status"`
	CarrierCode         string     `gorm:"size:32" json:"carrierCode"`
	CarrierName         string     `gorm:"size:64" json:"carrierName"`
	TrackingNo          string     `gorm:"size:64" json:"trackingNo"`
	ShippedAt           *time.Time `json:"shippedAt"`
	ExpectedArrivalDate *time.Time `json:"expectedArrivalDate"`
	DeliveredAt         *time.Time `json:"deliveredAt"`
	ShipFromAddressID   uint64     `json:"shipFromAddressId"`
	ReceiverName        string     `gorm:"size:64" json:"receiverName"`
	ReceiverPhone       string     `gorm:"size:32" json:"receiverPhone"`
	ReceiverAddress     string     `gorm:"size:255" json:"receiverAddress"`
	Remark              string     `gorm:"type:text" json:"remark"`
	CreatedAt           time.Time  `json:"createdAt"`
	UpdatedAt           time.Time  `json:"updatedAt"`
	Items               []DistShipmentItem `gorm:"foreignKey:ShipmentID" json:"items,omitempty"`
}

func (DistShipment) TableName() string { return "dist_shipments" }

type DistShipmentItem struct {
	ID         uint64    `gorm:"primaryKey" json:"id"`
	TenantID   uint64    `gorm:"index;not null" json:"tenantId"`
	ShipmentID uint64    `gorm:"index;not null" json:"shipmentId"`
	DistOrderItemID   uint64    `gorm:"index;not null" json:"distOrderItemId"`
	Qty        int       `gorm:"not null" json:"qty"`
	CreatedAt  time.Time `json:"createdAt"`
	UpdatedAt  time.Time `json:"updatedAt"`
}

func (DistShipmentItem) TableName() string { return "dist_shipment_items" }

type DistReceipt struct {
	ID            uint64     `gorm:"primaryKey" json:"id"`
	TenantID      uint64     `gorm:"index;not null" json:"tenantId"`
	DistOrderID          uint64     `gorm:"index;not null" json:"distOrderId"`
	PayAmount     float64    `gorm:"type:decimal(14,2);not null" json:"payAmount"`
	PayMethod     string     `gorm:"size:32" json:"payMethod"`
	PayAccount    string     `gorm:"size:128" json:"payAccount"`
	PayeeAccount  string     `gorm:"size:128" json:"payeeAccount"`
	PayeeName     string     `gorm:"size:128" json:"payeeName"`
	PayStatus     string     `gorm:"size:16;default:paid" json:"payStatus"`
	PaidAt        *time.Time `json:"paidAt"`
	Remark        string     `gorm:"type:text" json:"remark"`
	CreatedAt     time.Time  `json:"createdAt"`
	UpdatedAt     time.Time  `json:"updatedAt"`
}

func (DistReceipt) TableName() string { return "dist_receipts" }

type DistAttachment struct {
	ID         uint64    `gorm:"primaryKey" json:"id"`
	TenantID   uint64    `gorm:"index;not null" json:"tenantId"`
	DistOrderID       uint64    `gorm:"index;not null" json:"distOrderId"`
	PaymentID  uint64    `json:"paymentId"`
	ShipmentID uint64    `gorm:"index" json:"shipmentId"`
	FileType   string    `gorm:"size:32;not null" json:"fileType"`
	FileName   string    `gorm:"size:255;not null" json:"fileName"`
	FileURL    string    `gorm:"size:512;not null" json:"fileUrl"`
	UploadedBy uint64    `json:"uploadedBy"`
	Remark     string    `gorm:"type:text" json:"remark"`
	CreatedAt  time.Time `json:"createdAt"`
	UpdatedAt  time.Time `json:"updatedAt"`
}

func (DistAttachment) TableName() string { return "dist_attachments" }

File: web/src/api/tracking.ts
import client, { unwrap } from './client'

export interface ShipmentItem {
  id?: number
  distOrderItemId: number
  skuId?: number
  qty: number
}

export interface Shipment {
  id: number
  distOrderId: number
  shipmentNo: string
  status: string
  carrierCode?: string
  carrierName?: string
  trackingNo?: string
  shippedAt?: string
  expectedArrivalDate?: string
  deliveredAt?: string
  receiverName?: string
  receiverPhone?: string
  receiverAddress?: string
  remark?: string
  items: ShipmentItem[]
  createdAt: string
}

export interface Receipt {
  id: number
  distOrderId: number
  payAmount: number
  payMethod?: string
  payAccount?: string
  payeeAccount?: string
  payeeName?: string
  payStatus: string
  paidAt?: string
  remark?: string
  createdAt: string
}

export interface Attachment {
  id: number
  distOrderId: number
  paymentId?: number
  shipmentId?: number
  fileType: string
  fileName: string
  fileUrl: string
  uploadedBy?: number
  remark?: string
  createdAt: string
}

export const SHIPMENT_STATUS_MAP: Record<string, string> = {
  pending: '待发货',
  shipped: '已发货',
  in_transit: '运输中',
  delivered: '已签收',
  exception: '异常',
}

export const ATTACHMENT_TYPE_MAP: Record<string, string> = {
  dist_sales_order: '分销商销售单',
  payment_screenshot: '收款截图',
  shipment_photo: '物流发货照片',
  contract: '合同',
  other: '其他',
}

export async function fetchShipments(distOrderId: number) {
  return unwrap<Shipment[]>(await client.get(`/dist-orders/${distOrderId}/shipments`))
}

export async function createShipment(distOrderId: number, data: Partial<Shipment> & { items?: ShipmentItem[] }) {
  return unwrap<Shipment>(await client.post(`/dist-orders/${distOrderId}/shipments`, data))
}

export async function syncShipmentsFromOrders(distOrderId: number, refSoId?: number) {
  return unwrap<{ created: number; updated: number; skipped: number; errors?: string[] }>(
    await client.post(`/dist-orders/${distOrderId}/shipments/sync-from-orders`, refSoId ? { refSoId } : {}),
  )
}

export async function updateShipmentStatus(distOrderId: number, shipmentId: number, status: string) {
  return unwrap<Shipment>(await client.patch(`/dist-orders/${distOrderId}/shipments/${shipmentId}/status`, { status }))
}

export async function deleteShipment(distOrderId: number, shipmentId: number) {
  return unwrap(await client.delete(`/dist-orders/${distOrderId}/shipments/${shipmentId}`))
}

export async function fetchReceipts(distOrderId: number) {
  return unwrap<Receipt[]>(await client.get(`/dist-orders/${distOrderId}/receipts`))
}

export async function createReceipt(distOrderId: number, data: Partial<Receipt>) {
  return unwrap<Receipt>(await client.post(`/dist-orders/${distOrderId}/receipts`, data))
}

export async function deleteReceipt(distOrderId: number, receiptId: number) {
  return unwrap(await client.delete(`/dist-orders/${distOrderId}/receipts/${receiptId}`))
}

export async function fetchAttachments(distOrderId: number) {
  return unwrap<Attachment[]>(await client.get(`/dist-orders/${distOrderId}/attachments`))
}

export async function createAttachment(distOrderId: number, data: {
  fileType: string
  fileName: string
  fileUrl: string
  paymentId?: number
  shipmentId?: number
  remark?: string
}) {
  return unwrap<Attachment>(await client.post(`/dist-orders/${distOrderId}/attachments`, data))
}

export async function deleteAttachment(distOrderId: number, attachmentId: number) {
  return unwrap(await client.delete(`/dist-orders/${distOrderId}/attachments/${attachmentId}`))
}

export async function uploadFile(file: File) {
  const form = new FormData()
  form.append('file', file)
  const res = await client.post('/upload', form, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
  return unwrap<{ url: string; fileName: string }>(res)
}
